File: ts/step7-strategy-pattern/raceCar.ts
export class RaceCar {
    private name: string;
    private position: number;

    constructor(name: string) {
        this.name = name;
        this.position = 0;
    }

    public move(randomValue: number): void {
        if (randomValue >= 4) {
            this.position++;
        }
    }

    public getName(): string { return this.name; }
    public getPosition(): number { return this.position; }
}

File: java/step7-strategy-pattern/src/main/java/badcode/RaceCar.java
package badcode;

public class RaceCar {
    private String name;
    private int position;

    public RaceCar(String name) {
        this.name = name;
        this.position = 0;
    }

    public void move(int randomValue) {
        if (randomValue >= 4) {
            this.position++;
        }
    }

    public String getName() {
        return name;
    }

    public int getPosition() {
        return position;
    }
}
